opencog/persist/load_scm/fast_load.cpp

```diff
@@ -0,0 +1,172 @@
+#ifndef FAST_LOAD_CPP
+#define FAST_LOAD_CPP
+#include <opencog/atomspace/AtomSpace.h>
+#include <opencog/guile/SchemeEval.h>
+#include <opencog/atoms/atom_types/NameServer.h>
+#include "fast_load.h"
+
+using namespace opencog;
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <stdexcept>
+
+bool check = false;
+void get_next_expr(const std::string& s, int& l, int& r)
+{
+    int l1 = l;
+    while(s[l1] != '(' && l1 < r) {
+        if(s[l1] != ' ' && s[l1] != '\t' && s[l1] != '\n') {
+            throw std::runtime_error("Invalid syntax #1 in " + s.substr(l,r-l+1) + " at |" + s[l1] + "|");
+        }
+        l1++;
+    }
+    if(l1 >= r) {
+        l = l1;
+        return;
+    }
+    l = l1 + 1;
+    int count = 1;
+    bool par = false;
+    do {
+        l1++;
+        if(s[l1] == '"') par = !par;
+        if(par) continue;
+        if(s[l1] == '(') count++;
+        if(s[l1] == ')') count--;
+    } while(l1 <= r && count > 0);
+    if(count != 0) {
+        throw std::runtime_error("Invalid syntax #2 in " + s);
+    }
+    r = l1 - 1;
+}
+
+void get_next_token(const std::string& s, int& l, int& r)
```

Comment:
It would be nice to have a comment that explained what this function did.  I mean, it is not hard to tell by reading the source, but still, reading the source takes more work, more effort than reading comments.